plugins/components/collider/Collider.hpp

```diff
@@ -64,20 +74,26 @@ namespace Components {
          * @throws std::runtime_error If the data size is invalid.
          */
         void deserialize(std::vector<uint8_t> &data) override {
-            if (data.size() != getConstantSize())
+            if (data.size() != getSize())
                 throw std::runtime_error("Invalid data size for Collider component");
 
-            width = *reinterpret_cast<float *>(&data[0]);
-            height = *reinterpret_cast<float *>(&data[sizeof(float)]);
-        };
+            uint32_t widthBits = (data[0] << 24) | (data[1] << 16) | (data[2] << 8) | data[3];
+            uint32_t heightBits = (data[4] << 24) | (data[5] << 16) | (data[6] << 8) | data[7];
```

Comment:
Instead of bitshifting I think you could cast to uint32_t pointer: something like the following:

```cpp
uint32_t width_net = *((uint32_t *)&data[0]);
uint32_t height_net = *((uint32_t *)&data[4]);
```
You could use `reinterpret_cast` instead of C-style cast to lean more towards C++ style

Reply:
Gonna try this ^^, will report

Reply:
Indeed, i could have just copied from Position deserialize function since its the same...

Reply:
All done